static/src/javascripts-legacy/projects/commercial/modules/hosted/gallery.js

```diff
@@ -94,11 +94,19 @@ define([
     HostedGallery.prototype.initScroll = function () {
         bean.on(this.nextBtn, 'click', function(){
             this.scrollTo(this.index + 1);
-            this.trigger.bind(this, 'next', {nav: 'Click'});
+            if (this.index < this.$images.length) {
+                this.trigger('next', {nav: 'Click'});
+            } else {
+                this.trigger('reload');
+            }
         }.bind(this));
         bean.on(this.prevBtn, 'click', function(){
             this.scrollTo(this.index - 1);
-            this.trigger.bind(this, 'prev', {nav: 'Click'});
+            if (this.index > 1) {
+                this.trigger('prev', {nav: 'Click'});
+            } else {
+                this.trigger('reload');
+            }
         }.bind(this));
 
         bean.on(this.$scrollEl[0], 'scroll', throttle(this.fadeContent.bind(this), 20));
```

Comment:
Unrelated: you can use a [passive listener](https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md) and no longer throttle here instead, with the `common/utils/add-event-listener` module.